astropy/modeling/parameters.py

```diff
@@ -728,22 +728,22 @@ def __rtruediv__(self, val):
         return val / self.value
 
     def __eq__(self, val):
-        return (np.asarray(self) == np.asarray(val)).all()
+        return np.asanyarray(self) == np.asanyarray(val)
```

Comment:
Though there's no real reason to call methods like `__eq__` explicitly either.  I could just write `self.__array__() == other`.